src/uu/wc/src/wc.rs

```diff
@@ -214,11 +216,70 @@ fn word_count_from_path(path: &str, settings: &Settings) -> WcResult<WordCount>
     }
 }
 
+/// Compute the number of digits needed to represent all counts in all files.
+///
+/// `files` may include zero or more "-" entries, each of which
+/// represents reading from `stdin`. The presence of any "-" entry
+/// causes this function to return a width that is at least
+/// [`MINIMUM_WIDTH`].
+///
+/// If `files` is empty, then this function returns 1. If file
+/// metadata could not be read from any of the `files` and "-" is not
+/// an element of `files`, then this function returns 1.
+///
+/// # Errors
+///
+/// This function will return an error if there is a problem accessing
+/// the metadata of at least one of the named files.
+///
+/// # Examples
+///
+/// An empty slice implies a width of 1:
+///
+/// ```rust,ignore
+/// assert_eq!(1, max_width(&vec![]));
+/// ```
+///
+/// The presence of "-" implies a minimum width:
+///
+/// ```rust,ignore
+/// let files = vec![String::from("-")];
+/// assert_eq!(7, max_width(&files));
+/// ```
+fn max_width(files: &[String]) -> WcResult<usize> {
+    let mut result: usize = 1;
+    for file in files {
+        if file == "-" {
+            result = result.max(MINIMUM_WIDTH);
+            continue;
+        }
+        let path = Path::new(file);
+        let metadata = fs::metadata(path)?;
+        if !metadata.is_file() {
+            continue;
+        }
+        let num_bytes = metadata.len();
```

Comment:
This has the downside of computing the number of bytes in a file twice, once here and once during the processing of files. A future pull request can eliminate that redundancy.

Reply:
can you add a comment with TODO and create an issue about this ? thanks

Reply:
Yes, I will add a comment and create a new GitHub issue.

Reply:
The issue is #2201.